### ISSUE TITLE
Stackable cables

### ISSUE THREAD

**Issue by titovm** [VCVRack/Rack#57]
Adding multiple cables to one out/input would be also quite helpful, though further from the real world experience :)

**Comment by AndrewBelt** [comment]
This is a huge one. We need to decide on the following

- Should cables stack arbitrarily, as long as only one output is involved in the "group"? My opinion in *no*, UI would be too complicated and a mess. So input ports shouldn't stack.
- Should grabbing the port select the cable on the top of the stack?
- Should grabbing the port open a "ring menu" as in [some video games](https://cdn1.epicgames.com/ue/item/RingMenuConstKit_Screenshot_05-1920x1080-53b014fd0c25a7af8936b4b9a70662c0.png)? The steps would be: Click and hold the left mouse button on a port with multiple plugs, the jacks spread into a ring, drag your mouse to your desired cable, other plugs return to their original position and now you're currently grabbing a plug.

**Comment by synthi** [comment]
The "ring menu" is a fresh approach, I like it!

**Comment by ModeratelyModular** [comment]
I guess this is the right place to mention I could not find use for the scope since you can use outputs only once? But I'm not an expert (though I got it working (nicely!), using VCO for LFO and using the sequencer).

**Comment by titovm** [comment]
Thanks @AndrewBelt for title correction.

My guess is that coding a "ring menu" would be quite hard and involve unimportant libraries, unless you're up for it. On the other hand making input sockets exclusive and be able to drag as much cables as you want from one output like it's done in Softube's Modular would be easier to make and sooner to implement.

**Comment by synthi** [comment]
How do you choose what cable to remove in softube modular??

**Comment by titovm** [comment]
@synthi you just unplug the input, which is exclusive to one plug

**Comment by synthi** [comment]
@ModeratelyModular for now you can use multiples for sending one in to several outs

**Comment by ModeratelyModular** [comment]
@synthi Thanks for your response, I guesses so indeed, from the pictures, but: 

1. Getting the AudibleInstruments in VCV was somewhat (frustrating) cumbersome. I finally got them (don't ask me how), but haven't tried them out yet (it was late and i had too much fun with the sounds). 
2. Somehow it feels to me, when in the fundamentals, it should work with those (and all evening that was all I got).

**Comment by oblivionratula** [comment]
Wandering OT, but I agree @ModeratelyModular ,  a mult is pretty 'Fundamental.' Then again, a separate issue, a built-in mult on  a Scope is pretty handy/expected.

**Comment by dotasek** [comment]
I feel like grabbing the cable at the top of the stack is a good way to go. It's cumbersome, but it's cumbersome in the same way as actual cables are. As a plus, one click to one action is pretty intuitive to figure out almost immediately.

**Comment by AndrewBelt** [comment]
^ That's true, and you can always "grab" cables off the bottom of the stack by removing their inputs instead and creating a new cable.

**Comment by AndrewBelt** [comment]
Added with 7ee43d337d07af6ce72e52074fdd38417820a0ab
